src/App.tsx

```diff
@@ -15,36 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_ALPHABET = 'ALPHABET';
+const SORT_FIELD_LENGTH = 'LENGTH';
+
+type SortField = typeof SORT_FIELD_ALPHABET | typeof SORT_FIELD_LENGTH | '';
+
+function getPrepearedGoods(
+  goods: string[],
+  sortField: SortField,
+  reversed: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
```

Comment:
There is a typo in the function name `getPrepearedGoods`. It should be `getPreparedGoods` to match standard spelling conventions.